planning/behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -1904,31 +1949,30 @@ std::pair<bool, bool> GoalPlannerModule::isSafePath() const
       ego_predicted_path_params_, pull_over_path.points, current_pose, current_velocity,
       ego_seg_idx, is_object_front, limit_to_max_velocity);
 
-  // filtering objects with velocity, position and class
-  const auto filtered_objects = utils::path_safety_checker::filterObjects(
-    dynamic_object, route_handler, pull_over_lanes, current_pose.position,
-    objects_filtering_params_);
-
   // filtering objects based on the current position's lane
-  const auto target_objects_on_lane = utils::path_safety_checker::createTargetObjectsOnLane(
-    pull_over_lanes, route_handler, filtered_objects, objects_filtering_params_);
+  const auto expanded_pull_over_lanes_between_ego =
+    goal_planner_utils::generateBetweenEgoAndExpandedPullOverLanes(
+      pull_over_lanes, left_side_parking_, current_pose, planner_data_->parameters.vehicle_info,
+      parameters_->outer_road_detection_offset, parameters_->inner_road_detection_offset);
```

Comment:
how about use first road lane center pose in pull over lane segments as ego pose to fix https://github.com/autowarefoundation/autoware.universe/pull/6369/files#r1485547115?

```suggestion
  const Pose ego_pose_for_expand = std::invoke([&]() {
    // get first road lane in pull over lanes segment
    const auto fist_road_lane = std::invoke([&]() {
      const auto first_pull_over_lane = pull_over_lanes.front();
      if (!route_handler->isShoulderLanelet(first_pull_over_lane)) {
        return first_pull_over_lane;
      }
      const auto road_lane_opt = left_side_parking_
                                   ? route_handler->getRightLanelet(first_pull_over_lane)
                                   : route_handler->getLeftLanelet(first_pull_over_lane);
      if (road_lane_opt) {
        return road_lane_opt.value();
      }
      return first_pull_over_lane;
    });
    // generate first road lane pose
    Pose first_road_pose{};
    const auto first_road_point =
      lanelet::utils::conversion::toGeomMsgPt(fist_road_lane.centerline().front());
    const double lane_yaw = lanelet::utils::getLaneletAngle(fist_road_lane, first_road_point);
    first_road_pose.position = first_road_point;
    first_road_pose.orientation = tier4_autoware_utils::createQuaternionFromYaw(lane_yaw);
    // if current ego pose is before pull over lanes segment, use first road lanelet center pose
    if (
      calcSignedArcLength(pull_over_path.points, first_road_pose.position, current_pose.position) <
      0.0) {
      return first_road_pose;
    }
    // if current ego pose is in pull over lanes segment, use current ego pose
    return current_pose;
  });

  const auto expanded_pull_over_lanes_between_ego =
    goal_planner_utils::generateBetweenEgoAndExpandedPullOverLanes(
      pull_over_lanes, left_side_parking_, ego_pose_for_expand,
      planner_data_->parameters.vehicle_info, parameters_->outer_road_detection_offset,
      parameters_->inner_road_detection_offset);
```

Reply:
this my lamda is too long so may be better to separate functions